src/ints/mouse_dos_driver.cpp

```diff
@@ -0,0 +1,1440 @@
+            INT10_GetPixel(static_cast<uint16_t>(x),
+                           static_cast<uint16_t>(y),
+                           state.page,
+                           &state.backData[dataPos++]);
+        };
+        dataPos = static_cast<uint16_t>(dataPos + addx2);
+    };
+    state.background = true;
+    state.backposx = static_cast<uint16_t>(GetPosX() / xratio - state.hot_x);
+    state.backposy = static_cast<uint16_t>(GetPosY() - state.hot_y);
+
+    // Draw Mousecursor
+    dataPos = static_cast<uint16_t>(addy * CURSOR_SIZE_X);
+    const auto screenMask = state.user_screen_mask ? state.user_def_screen_mask
+                                                   : DEFAULT_SCREEN_MASK;
+    const auto cursorMask = state.user_cursor_mask ? state.user_def_cursor_mask
+                                                   : DEFAULT_CURSOR_MASK;
+    for (y = y1; y <= y2; y++) {
+        uint16_t scMask = screenMask[addy + y - y1];
+        uint16_t cuMask = cursorMask[addy + y - y1];
+        if (addx1 > 0) {
+            scMask  = static_cast<uint16_t>(scMask << addx1);
+            cuMask  = static_cast<uint16_t>(cuMask << addx1);
+            dataPos = static_cast<uint16_t>(dataPos + addx1);
+        };
+        for (x = x1; x <= x2; x++) {
+            constexpr auto HIGHESTBIT = (1 << (CURSOR_SIZE_X - 1));
+            uint8_t pixel             = 0;
+            // ScreenMask
+            if (scMask & HIGHESTBIT)
+                pixel = state.backData[dataPos];
+            scMask = static_cast<uint16_t>(scMask << 1);
+            // CursorMask
+            if (cuMask & HIGHESTBIT)
+                pixel = pixel ^ 0x0f;
+            cuMask = static_cast<uint16_t>(cuMask << 1);
+            // Set Pixel
+            INT10_PutPixel(static_cast<uint16_t>(x),
+                           static_cast<uint16_t>(y),
+                           state.page,
+                           pixel);
+            dataPos++;
+        };
+        dataPos = static_cast<uint16_t>(addx2 + dataPos);
+    };
+
+    RestoreVgaRegisters();
+}
+
+// ***************************************************************************
+// DOS driver interface implementation
+// ***************************************************************************
+
+static void UpdateDriverActive()
+{
+    mouse_shared.active_dos = (state.sub_mask != 0);
+    MOUSE_NotifyStateChanged();
+}
+
+static uint8_t GetResetWheel8bit()
+{
+    if (!state.cute_mouse) // wheel requires CuteMouse extensions
+        return 0;
+
+    const int16_t tmp = static_cast<int16_t>(
+            std::clamp(wheel,
+                       static_cast<int16_t>(INT8_MIN),
+                       static_cast<int16_t>(INT8_MAX)));
+    wheel = 0; // reading always int8_t the wheel counter
+
+    // 0xff for -1, 0xfe for -2, etc.
+    return static_cast<uint8_t>((tmp >= 0) ? tmp : (0x100 + tmp));
+}
+
+static uint16_t GetResetWheel16bit()
+{
+    if (!state.cute_mouse) // wheel requires CuteMouse extensions
```

Comment:
Just an idea: can't we integrate CuteMouse into the codebase? Seems like lots of this code is tightly coupled to it, and most likely CuteMouse will never see any updates (no reason for it, right? it just works).

Reply:
1) CuteMouse is written in x86 assembler. 2) Contrary to DOSBox internal driver, it needs some conventional memory. 3) If CuteMouse is running, seamless mouse integration is not possible.

So this is not a perfect replacement.

Reply:
Okay, that makes sense, we should leave it as it is then.